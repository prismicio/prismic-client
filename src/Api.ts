import Predicates from './Predicates';
import { RequestHandler, RequestCallback } from './request';
import { Document } from './documents';
import { ApiCache } from './cache';
import ResolvedApi, { ApiData } from './ResolvedApi';
import HttpClient from './HttpClient';

export interface ApiOptions {
  accessToken?: string;
  requestHandler?: RequestHandler;
  req?: any;
  apiCache?: ApiCache;
  apiDataTTL?: number;
  proxyAgent?: any;
}

export default class PrismicApi {
  url: string;
  options: ApiOptions;
  apiDataTTL: number;
  httpClient: HttpClient;

  constructor(url: string, options?: ApiOptions) {
    this.options = options || {};
    this.url = url;
    if (this.options.accessToken) {
      const accessTokenParam = `access_token=${this.options.accessToken}`;
      this.url += (url.indexOf('?') > -1 ? '&' : '?') + accessTokenParam;
    }
    this.apiDataTTL = this.options.apiDataTTL || 5;
    this.httpClient = new HttpClient(
      this.options.requestHandler,
      this.options.apiCache,
      this.options.proxyAgent,
    );
  }

  /**
   * Fetches data used to construct the api client, from cache if it's
   * present, otherwise from calling the prismic api endpoint (which is
   * then cached).
   */
  get(cb?: RequestCallback<ResolvedApi>): Promise<ResolvedApi> {
    return this.httpClient.cachedRequest<ApiData>(this.url, { ttl: this.apiDataTTL }).then((data) => {
      const resolvedApi = new ResolvedApi(data, this.httpClient, this.options);
      cb && cb(null, resolvedApi);
      return resolvedApi;
    }).catch((error) => {
      cb && cb(error);
      throw error;
    });
  }
}
