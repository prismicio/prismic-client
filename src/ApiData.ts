export interface QuickRoutes {
  url: string;
  lastUpdate: number;
  enabled: boolean;
}

export interface ApiData {
  refs: Ref[];
  bookmarks: { [key: string]: string };
  types: { [key: string]: string };
  tags: string[];
  forms: { [key: string]: Form };
  quickRoutes: QuickRoutes;
  experiments: any;
  oauth_initiate: string;
  oauth_token: string;
  version: string;
  licence: string;
}
