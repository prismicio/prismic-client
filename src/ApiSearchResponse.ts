import { Document } from "./documents";

export default interface ApiSearchResponse<T = any> {
  page: number;
  results_per_page: number;
  results_size: number;
  total_results_size: number;
  total_pages: number;
  next_page: string;
  prev_page: string;
  results: Document<T>[];
}
