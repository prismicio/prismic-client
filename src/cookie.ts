/**
 * The well-known name of the cookie used to store a Prismic preview session's
 * ref.
 */
export const preview = "io.prismic.preview"
