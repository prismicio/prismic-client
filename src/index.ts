// Primary library API.
import { getRepositoryEndpoint } from "./getRepositoryEndpoint";
export { getRepositoryEndpoint };
/**
 * @deprecated Renamed to `getRepositoryEndpoint`.
 */
// TODO: Remove in v3.
export const getEndpoint = getRepositoryEndpoint;
export { getRepositoryName } from "./getRepositoryName";
export { getGraphQLEndpoint } from "./getGraphQLEndpoint";
export { isRepositoryName } from "./isRepositoryName";
export { isRepositoryEndpoint } from "./isRepositoryEndpoint";
export { buildQueryURL } from "./buildQueryURL";
export { createClient, Client } from "./client";

// Predicates API.
import { predicate } from "./predicate";
export { predicate };
/**
 * @deprecated Renamed to `predicate` (without an "s").
 */
// TODO: Remove in v3.
export const predicates = predicate;
/**
 * @deprecated Renamed to `predicate` (lowercase and without an "s").
 */
// TODO: Remove in v3.
export const Predicates = predicate;

// Custom errors used by Client.
export { PrismicError } from "./PrismicError";
export { ForbiddenError } from "./ForbiddenError";
export { ParsingError } from "./ParsingError";
export { NotFoundError } from "./NotFoundError";

// A collection of well-known cookie names shared between Prismic libraries and systems.
export * as cookie from "./cookie";

// General types used throughout the project. These are made public to allow users to better type their projects.
export type { CreateClient, ClientConfig } from "./client";
export type { QueryParams, BuildQueryURLArgs } from "./buildQueryURL";
export type {
	AbortSignalLike,
	FetchLike,
	HttpRequestLike,
	Ordering,
	RequestInitLike,
	ResponseLike,
	Route,
} from "./types";
