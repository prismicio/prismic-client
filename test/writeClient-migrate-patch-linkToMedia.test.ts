import { testMigrationFieldPatching } from "./__testutils__/testMigrationFieldPatching"

import type {
	InjectMigrationSpecificTypes,
	LinkToMediaField,
	RichTextField,
} from "../src"
import { LinkType, RichTextNodeType } from "../src"
import type { Asset } from "../src/types/api/asset/asset"
import type { MigrationLinkToMedia } from "../src/types/migration/Asset"

const assetToLinkToMedia = (
	asset: Asset,
	text?: string,
): LinkToMediaField<"filled"> => {
	return {
		id: asset.id,
		link_type: LinkType.Media,
		name: asset.filename,
		kind: asset.kind,
		url: asset.url,
		size: `${asset.size}`,
		height: typeof asset.height === "number" ? `${asset.height}` : undefined,
		width: typeof asset.width === "number" ? `${asset.width}` : undefined,
		// TODO: Remove when link text PR is merged
		// @ts-expect-error - Future-proofing for link text
		text,
	}
}

testMigrationFieldPatching<
	| MigrationLinkToMedia
	| LinkToMediaField
	| InjectMigrationSpecificTypes<RichTextField>
>("patches link to media fields", {
	new: ({ migration }) => {
		return {
			link_type: LinkType.Media,
			id: migration.createAsset("foo", "foo.png"),
		}
	},
	newWithText: ({ migration }) => {
		return {
			link_type: LinkType.Media,
			id: migration.createAsset("foo", "foo.png"),
			text: "foo",
		}
	},
	newNonImage: ({ migration }) => {
		const migrationAsset = migration.createAsset("foo", "foo.pdf", {
			tags: ["__pdf__"],
		})

		return {
			link_type: LinkType.Media,
			id: migrationAsset,
		}
	},
	existing: ({ existingAssets }) => assetToLinkToMedia(existingAssets[0]),
	empty: ({ migration }) => {
		const asset = migration.createAsset("foo", "foo.png")
		asset.config.id = "empty"

		return {
			link_type: LinkType.Media,
			id: asset,
		}
	},
	richTextNew: ({ migration }) => [
		{
			type: RichTextNodeType.paragraph,
			text: "lorem",
			spans: [
				{ type: RichTextNodeType.strong, start: 0, end: 5 },
				{
					type: RichTextNodeType.hyperlink,
					start: 0,
					end: 5,
					data: {
						link_type: LinkType.Media,
						id: migration.createAsset("foo", "foo.png"),
					},
				},
			],
		},
	],
	richTextExisting: ({ existingAssets }) => [
		{
			type: RichTextNodeType.paragraph,
			text: "lorem",
			spans: [
				{ type: RichTextNodeType.strong, start: 0, end: 5 },
				{
					type: RichTextNodeType.hyperlink,
					start: 0,
					end: 5,
					data: assetToLinkToMedia(existingAssets[0]),
				},
			],
		},
	],
})

testMigrationFieldPatching<LinkToMediaField | RichTextField>(
	"patches link to media fields (from Prismic)",
	{
		simple: ({ ctx, mockedDomain }) => {
			return {
				...ctx.mock.value.linkToMedia({ state: "filled" }),
				id: "foo-id",
				url: `${mockedDomain}/foo.png`,
			}
		},
		inRichText: ({ ctx, mockedDomain }) => [
			{
				type: RichTextNodeType.paragraph,
				text: "lorem",
				spans: [
					{ type: RichTextNodeType.strong, start: 0, end: 5 },
					{
						type: RichTextNodeType.hyperlink,
						start: 0,
						end: 5,
						data: {
							...ctx.mock.value.linkToMedia({ state: "filled" }),
							id: "foo-id",
							url: `${mockedDomain}/foo.png`,
						},
					},
				],
			},
		],
	},
	{ mode: "fromPrismic" },
)
