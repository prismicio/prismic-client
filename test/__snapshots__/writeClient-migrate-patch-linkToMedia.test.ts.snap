// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`patches link to media fields (from Prismic) > inRichText > group 1`] = `
{
  "group": [
    {
      "field": [
        {
          "spans": [
            {
              "end": 5,
              "start": 0,
              "type": "strong",
            },
            {
              "data": {
                "id": "fc26fe8d098",
                "link_type": "Media",
              },
              "end": 5,
              "start": 0,
              "type": "hyperlink",
            },
          ],
          "text": "lorem",
          "type": "paragraph",
        },
      ],
    },
  ],
}
`;

exports[`patches link to media fields (from Prismic) > inRichText > shared slice 1`] = `
{
  "slices": [
    {
      "id": "9b9dd0f2c3f",
      "items": [
        {
          "field": [
            {
              "spans": [
                {
                  "end": 5,
                  "start": 0,
                  "type": "strong",
                },
                {
                  "data": {
                    "id": "961adcf1f5d",
                    "link_type": "Media",
                  },
                  "end": 5,
                  "start": 0,
                  "type": "hyperlink",
                },
              ],
              "text": "lorem",
              "type": "paragraph",
            },
          ],
        },
      ],
      "primary": {
        "field": [
          {
            "spans": [
              {
                "end": 5,
                "start": 0,
                "type": "strong",
              },
              {
                "data": {
                  "id": "961adcf1f5d",
                  "link_type": "Media",
                },
                "end": 5,
                "start": 0,
                "type": "hyperlink",
              },
            ],
            "text": "lorem",
            "type": "paragraph",
          },
        ],
        "group": [
          {
            "field": [
              {
                "spans": [
                  {
                    "end": 5,
                    "start": 0,
                    "type": "strong",
                  },
                  {
                    "data": {
                      "id": "961adcf1f5d",
                      "link_type": "Media",
                    },
                    "end": 5,
                    "start": 0,
                    "type": "hyperlink",
                  },
                ],
                "text": "lorem",
                "type": "paragraph",
              },
            ],
          },
        ],
      },
      "slice_label": null,
      "slice_type": "nunc",
      "variation": "ullamcorper",
      "version": "8bfc905",
    },
  ],
}
`;

exports[`patches link to media fields (from Prismic) > inRichText > slice 1`] = `
{
  "slices": [
    {
      "id": "5306297c5ed",
      "items": [
        {
          "field": [
            {
              "spans": [
                {
                  "end": 5,
                  "start": 0,
                  "type": "strong",
                },
                {
                  "data": {
                    "id": "54f4a69ff72",
                    "link_type": "Media",
                  },
                  "end": 5,
                  "start": 0,
                  "type": "hyperlink",
                },
              ],
              "text": "lorem",
              "type": "paragraph",
            },
          ],
        },
      ],
      "primary": {
        "field": [
          {
            "spans": [
              {
                "end": 5,
                "start": 0,
                "type": "strong",
              },
              {
                "data": {
                  "id": "54f4a69ff72",
                  "link_type": "Media",
                },
                "end": 5,
                "start": 0,
                "type": "hyperlink",
              },
            ],
            "text": "lorem",
            "type": "paragraph",
          },
        ],
      },
      "slice_label": "Vel",
      "slice_type": "hac_habitasse",
    },
  ],
}
`;

exports[`patches link to media fields (from Prismic) > inRichText > static zone 1`] = `
{
  "field": [
    {
      "spans": [
        {
          "end": 5,
          "start": 0,
          "type": "strong",
        },
        {
          "data": {
            "id": "1d17b04a95c",
            "link_type": "Media",
          },
          "end": 5,
          "start": 0,
          "type": "hyperlink",
        },
      ],
      "text": "lorem",
      "type": "paragraph",
    },
  ],
}
`;

exports[`patches link to media fields (from Prismic) > simple > group 1`] = `
{
  "group": [
    {
      "field": {
        "id": "fc26fe8d098",
        "link_type": "Media",
      },
    },
  ],
}
`;

exports[`patches link to media fields (from Prismic) > simple > shared slice 1`] = `
{
  "slices": [
    {
      "id": "9b9dd0f2c3f",
      "items": [
        {
          "field": {
            "id": "961adcf1f5d",
            "link_type": "Media",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "961adcf1f5d",
          "link_type": "Media",
        },
        "group": [
          {
            "field": {
              "id": "961adcf1f5d",
              "link_type": "Media",
            },
          },
        ],
      },
      "slice_label": null,
      "slice_type": "nunc",
      "variation": "ullamcorper",
      "version": "8bfc905",
    },
  ],
}
`;

exports[`patches link to media fields (from Prismic) > simple > slice 1`] = `
{
  "slices": [
    {
      "id": "5306297c5ed",
      "items": [
        {
          "field": {
            "id": "54f4a69ff72",
            "link_type": "Media",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "54f4a69ff72",
          "link_type": "Media",
        },
      },
      "slice_label": "Vel",
      "slice_type": "hac_habitasse",
    },
  ],
}
`;

exports[`patches link to media fields (from Prismic) > simple > static zone 1`] = `
{
  "field": {
    "id": "1d17b04a95c",
    "link_type": "Media",
  },
}
`;

exports[`patches link to media fields > empty > group 1`] = `
{
  "group": [
    {
      "field": {
        "link_type": "Media",
      },
    },
  ],
}
`;

exports[`patches link to media fields > empty > shared slice 1`] = `
{
  "slices": [
    {
      "id": "9b9dd0f2c3f",
      "items": [
        {
          "field": {
            "link_type": "Media",
          },
        },
      ],
      "primary": {
        "field": {
          "link_type": "Media",
        },
        "group": [
          {
            "field": {
              "link_type": "Media",
            },
          },
        ],
      },
      "slice_label": null,
      "slice_type": "nunc",
      "variation": "ullamcorper",
      "version": "8bfc905",
    },
  ],
}
`;

exports[`patches link to media fields > empty > slice 1`] = `
{
  "slices": [
    {
      "id": "5306297c5ed",
      "items": [
        {
          "field": {
            "link_type": "Media",
          },
        },
      ],
      "primary": {
        "field": {
          "link_type": "Media",
        },
      },
      "slice_label": "Vel",
      "slice_type": "hac_habitasse",
    },
  ],
}
`;

exports[`patches link to media fields > empty > static zone 1`] = `
{
  "field": {
    "link_type": "Media",
  },
}
`;

exports[`patches link to media fields > existing > group 1`] = `
{
  "group": [
    {
      "field": {
        "height": "1",
        "id": "asset.id-existing",
        "kind": "image",
        "link_type": "Media",
        "name": "default.jpg",
        "size": "1",
        "url": "https://images.unsplash.com/photo-1470770903676-69b98201ea1c?w=4554&h=3036&fit=crop",
        "width": "1",
      },
    },
  ],
}
`;

exports[`patches link to media fields > existing > shared slice 1`] = `
{
  "slices": [
    {
      "id": "9b9dd0f2c3f",
      "items": [
        {
          "field": {
            "height": "1",
            "id": "asset.id-existing",
            "kind": "image",
            "link_type": "Media",
            "name": "default.jpg",
            "size": "1",
            "url": "https://images.unsplash.com/photo-1504567961542-e24d9439a724?w=4608&h=3456&fit=crop",
            "width": "1",
          },
        },
      ],
      "primary": {
        "field": {
          "height": "1",
          "id": "asset.id-existing",
          "kind": "image",
          "link_type": "Media",
          "name": "default.jpg",
          "size": "1",
          "url": "https://images.unsplash.com/photo-1504567961542-e24d9439a724?w=4608&h=3456&fit=crop",
          "width": "1",
        },
        "group": [
          {
            "field": {
              "height": "1",
              "id": "asset.id-existing",
              "kind": "image",
              "link_type": "Media",
              "name": "default.jpg",
              "size": "1",
              "url": "https://images.unsplash.com/photo-1504567961542-e24d9439a724?w=4608&h=3456&fit=crop",
              "width": "1",
            },
          },
        ],
      },
      "slice_label": null,
      "slice_type": "nunc",
      "variation": "ullamcorper",
      "version": "8bfc905",
    },
  ],
}
`;

exports[`patches link to media fields > existing > slice 1`] = `
{
  "slices": [
    {
      "id": "5306297c5ed",
      "items": [
        {
          "field": {
            "height": "1",
            "id": "asset.id-existing",
            "kind": "image",
            "link_type": "Media",
            "name": "default.jpg",
            "size": "1",
            "url": "https://images.unsplash.com/photo-1587502537745-84b86da1204f?w=6550&h=4367&fit=crop",
            "width": "1",
          },
        },
      ],
      "primary": {
        "field": {
          "height": "1",
          "id": "asset.id-existing",
          "kind": "image",
          "link_type": "Media",
          "name": "default.jpg",
          "size": "1",
          "url": "https://images.unsplash.com/photo-1587502537745-84b86da1204f?w=6550&h=4367&fit=crop",
          "width": "1",
        },
      },
      "slice_label": "Vel",
      "slice_type": "hac_habitasse",
    },
  ],
}
`;

exports[`patches link to media fields > existing > static zone 1`] = `
{
  "field": {
    "height": "1",
    "id": "asset.id-existing",
    "kind": "image",
    "link_type": "Media",
    "name": "default.jpg",
    "size": "1",
    "url": "https://images.unsplash.com/photo-1604537529428-15bcbeecfe4d?w=4240&h=2832&fit=crop",
    "width": "1",
  },
}
`;

exports[`patches link to media fields > new > group 1`] = `
{
  "group": [
    {
      "field": {
        "id": "fdd322ca9d5",
        "link_type": "Media",
      },
    },
  ],
}
`;

exports[`patches link to media fields > new > shared slice 1`] = `
{
  "slices": [
    {
      "id": "9b9dd0f2c3f",
      "items": [
        {
          "field": {
            "id": "c0eddbd0255",
            "link_type": "Media",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "c0eddbd0255",
          "link_type": "Media",
        },
        "group": [
          {
            "field": {
              "id": "c0eddbd0255",
              "link_type": "Media",
            },
          },
        ],
      },
      "slice_label": null,
      "slice_type": "nunc",
      "variation": "ullamcorper",
      "version": "8bfc905",
    },
  ],
}
`;

exports[`patches link to media fields > new > slice 1`] = `
{
  "slices": [
    {
      "id": "5306297c5ed",
      "items": [
        {
          "field": {
            "id": "82beb55ec2f",
            "link_type": "Media",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "82beb55ec2f",
          "link_type": "Media",
        },
      },
      "slice_label": "Vel",
      "slice_type": "hac_habitasse",
    },
  ],
}
`;

exports[`patches link to media fields > new > static zone 1`] = `
{
  "field": {
    "id": "bad670dad77",
    "link_type": "Media",
  },
}
`;

exports[`patches link to media fields > newNonImage > group 1`] = `
{
  "group": [
    {
      "field": {
        "id": "fdd322ca9d5",
        "link_type": "Media",
      },
    },
  ],
}
`;

exports[`patches link to media fields > newNonImage > shared slice 1`] = `
{
  "slices": [
    {
      "id": "9b9dd0f2c3f",
      "items": [
        {
          "field": {
            "id": "c0eddbd0255",
            "link_type": "Media",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "c0eddbd0255",
          "link_type": "Media",
        },
        "group": [
          {
            "field": {
              "id": "c0eddbd0255",
              "link_type": "Media",
            },
          },
        ],
      },
      "slice_label": null,
      "slice_type": "nunc",
      "variation": "ullamcorper",
      "version": "8bfc905",
    },
  ],
}
`;

exports[`patches link to media fields > newNonImage > slice 1`] = `
{
  "slices": [
    {
      "id": "5306297c5ed",
      "items": [
        {
          "field": {
            "id": "82beb55ec2f",
            "link_type": "Media",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "82beb55ec2f",
          "link_type": "Media",
        },
      },
      "slice_label": "Vel",
      "slice_type": "hac_habitasse",
    },
  ],
}
`;

exports[`patches link to media fields > newNonImage > static zone 1`] = `
{
  "field": {
    "id": "bad670dad77",
    "link_type": "Media",
  },
}
`;

exports[`patches link to media fields > newWithText > group 1`] = `
{
  "group": [
    {
      "field": {
        "id": "fdd322ca9d5",
        "link_type": "Media",
        "text": "foo",
      },
    },
  ],
}
`;

exports[`patches link to media fields > newWithText > shared slice 1`] = `
{
  "slices": [
    {
      "id": "9b9dd0f2c3f",
      "items": [
        {
          "field": {
            "id": "c0eddbd0255",
            "link_type": "Media",
            "text": "foo",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "c0eddbd0255",
          "link_type": "Media",
          "text": "foo",
        },
        "group": [
          {
            "field": {
              "id": "c0eddbd0255",
              "link_type": "Media",
              "text": "foo",
            },
          },
        ],
      },
      "slice_label": null,
      "slice_type": "nunc",
      "variation": "ullamcorper",
      "version": "8bfc905",
    },
  ],
}
`;

exports[`patches link to media fields > newWithText > slice 1`] = `
{
  "slices": [
    {
      "id": "5306297c5ed",
      "items": [
        {
          "field": {
            "id": "82beb55ec2f",
            "link_type": "Media",
            "text": "foo",
          },
        },
      ],
      "primary": {
        "field": {
          "id": "82beb55ec2f",
          "link_type": "Media",
          "text": "foo",
        },
      },
      "slice_label": "Vel",
      "slice_type": "hac_habitasse",
    },
  ],
}
`;

exports[`patches link to media fields > newWithText > static zone 1`] = `
{
  "field": {
    "id": "bad670dad77",
    "link_type": "Media",
    "text": "foo",
  },
}
`;

exports[`patches link to media fields > richTextExisting > group 1`] = `
{
  "group": [
    {
      "field": [
        {
          "spans": [
            {
              "end": 5,
              "start": 0,
              "type": "strong",
            },
            {
              "data": {
                "height": "1",
                "id": "asset.id-existing",
                "kind": "image",
                "link_type": "Media",
                "name": "default.jpg",
                "size": "1",
                "url": "https://images.unsplash.com/photo-1470770903676-69b98201ea1c?w=4554&h=3036&fit=crop",
                "width": "1",
              },
              "end": 5,
              "start": 0,
              "type": "hyperlink",
            },
          ],
          "text": "lorem",
          "type": "paragraph",
        },
      ],
    },
  ],
}
`;

exports[`patches link to media fields > richTextExisting > shared slice 1`] = `
{
  "slices": [
    {
      "id": "9b9dd0f2c3f",
      "items": [
        {
          "field": [
            {
              "spans": [
                {
                  "end": 5,
                  "start": 0,
                  "type": "strong",
                },
                {
                  "data": {
                    "height": "1",
                    "id": "asset.id-existing",
                    "kind": "image",
                    "link_type": "Media",
                    "name": "default.jpg",
                    "size": "1",
                    "url": "https://images.unsplash.com/photo-1504567961542-e24d9439a724?w=4608&h=3456&fit=crop",
                    "width": "1",
                  },
                  "end": 5,
                  "start": 0,
                  "type": "hyperlink",
                },
              ],
              "text": "lorem",
              "type": "paragraph",
            },
          ],
        },
      ],
      "primary": {
        "field": [
          {
            "spans": [
              {
                "end": 5,
                "start": 0,
                "type": "strong",
              },
              {
                "data": {
                  "height": "1",
                  "id": "asset.id-existing",
                  "kind": "image",
                  "link_type": "Media",
                  "name": "default.jpg",
                  "size": "1",
                  "url": "https://images.unsplash.com/photo-1504567961542-e24d9439a724?w=4608&h=3456&fit=crop",
                  "width": "1",
                },
                "end": 5,
                "start": 0,
                "type": "hyperlink",
              },
            ],
            "text": "lorem",
            "type": "paragraph",
          },
        ],
        "group": [
          {
            "field": [
              {
                "spans": [
                  {
                    "end": 5,
                    "start": 0,
                    "type": "strong",
                  },
                  {
                    "data": {
                      "height": "1",
                      "id": "asset.id-existing",
                      "kind": "image",
                      "link_type": "Media",
                      "name": "default.jpg",
                      "size": "1",
                      "url": "https://images.unsplash.com/photo-1504567961542-e24d9439a724?w=4608&h=3456&fit=crop",
                      "width": "1",
                    },
                    "end": 5,
                    "start": 0,
                    "type": "hyperlink",
                  },
                ],
                "text": "lorem",
                "type": "paragraph",
              },
            ],
          },
        ],
      },
      "slice_label": null,
      "slice_type": "nunc",
      "variation": "ullamcorper",
      "version": "8bfc905",
    },
  ],
}
`;

exports[`patches link to media fields > richTextExisting > slice 1`] = `
{
  "slices": [
    {
      "id": "5306297c5ed",
      "items": [
        {
          "field": [
            {
              "spans": [
                {
                  "end": 5,
                  "start": 0,
                  "type": "strong",
                },
                {
                  "data": {
                    "height": "1",
                    "id": "asset.id-existing",
                    "kind": "image",
                    "link_type": "Media",
                    "name": "default.jpg",
                    "size": "1",
                    "url": "https://images.unsplash.com/photo-1587502537745-84b86da1204f?w=6550&h=4367&fit=crop",
                    "width": "1",
                  },
                  "end": 5,
                  "start": 0,
                  "type": "hyperlink",
                },
              ],
              "text": "lorem",
              "type": "paragraph",
            },
          ],
        },
      ],
      "primary": {
        "field": [
          {
            "spans": [
              {
                "end": 5,
                "start": 0,
                "type": "strong",
              },
              {
                "data": {
                  "height": "1",
                  "id": "asset.id-existing",
                  "kind": "image",
                  "link_type": "Media",
                  "name": "default.jpg",
                  "size": "1",
                  "url": "https://images.unsplash.com/photo-1587502537745-84b86da1204f?w=6550&h=4367&fit=crop",
                  "width": "1",
                },
                "end": 5,
                "start": 0,
                "type": "hyperlink",
              },
            ],
            "text": "lorem",
            "type": "paragraph",
          },
        ],
      },
      "slice_label": "Vel",
      "slice_type": "hac_habitasse",
    },
  ],
}
`;

exports[`patches link to media fields > richTextExisting > static zone 1`] = `
{
  "field": [
    {
      "spans": [
        {
          "end": 5,
          "start": 0,
          "type": "strong",
        },
        {
          "data": {
            "height": "1",
            "id": "asset.id-existing",
            "kind": "image",
            "link_type": "Media",
            "name": "default.jpg",
            "size": "1",
            "url": "https://images.unsplash.com/photo-1604537529428-15bcbeecfe4d?w=4240&h=2832&fit=crop",
            "width": "1",
          },
          "end": 5,
          "start": 0,
          "type": "hyperlink",
        },
      ],
      "text": "lorem",
      "type": "paragraph",
    },
  ],
}
`;

exports[`patches link to media fields > richTextNew > group 1`] = `
{
  "group": [
    {
      "field": [
        {
          "spans": [
            {
              "end": 5,
              "start": 0,
              "type": "strong",
            },
            {
              "data": {
                "id": "fdd322ca9d5",
                "link_type": "Media",
              },
              "end": 5,
              "start": 0,
              "type": "hyperlink",
            },
          ],
          "text": "lorem",
          "type": "paragraph",
        },
      ],
    },
  ],
}
`;

exports[`patches link to media fields > richTextNew > shared slice 1`] = `
{
  "slices": [
    {
      "id": "9b9dd0f2c3f",
      "items": [
        {
          "field": [
            {
              "spans": [
                {
                  "end": 5,
                  "start": 0,
                  "type": "strong",
                },
                {
                  "data": {
                    "id": "c0eddbd0255",
                    "link_type": "Media",
                  },
                  "end": 5,
                  "start": 0,
                  "type": "hyperlink",
                },
              ],
              "text": "lorem",
              "type": "paragraph",
            },
          ],
        },
      ],
      "primary": {
        "field": [
          {
            "spans": [
              {
                "end": 5,
                "start": 0,
                "type": "strong",
              },
              {
                "data": {
                  "id": "c0eddbd0255",
                  "link_type": "Media",
                },
                "end": 5,
                "start": 0,
                "type": "hyperlink",
              },
            ],
            "text": "lorem",
            "type": "paragraph",
          },
        ],
        "group": [
          {
            "field": [
              {
                "spans": [
                  {
                    "end": 5,
                    "start": 0,
                    "type": "strong",
                  },
                  {
                    "data": {
                      "id": "c0eddbd0255",
                      "link_type": "Media",
                    },
                    "end": 5,
                    "start": 0,
                    "type": "hyperlink",
                  },
                ],
                "text": "lorem",
                "type": "paragraph",
              },
            ],
          },
        ],
      },
      "slice_label": null,
      "slice_type": "nunc",
      "variation": "ullamcorper",
      "version": "8bfc905",
    },
  ],
}
`;

exports[`patches link to media fields > richTextNew > slice 1`] = `
{
  "slices": [
    {
      "id": "5306297c5ed",
      "items": [
        {
          "field": [
            {
              "spans": [
                {
                  "end": 5,
                  "start": 0,
                  "type": "strong",
                },
                {
                  "data": {
                    "id": "82beb55ec2f",
                    "link_type": "Media",
                  },
                  "end": 5,
                  "start": 0,
                  "type": "hyperlink",
                },
              ],
              "text": "lorem",
              "type": "paragraph",
            },
          ],
        },
      ],
      "primary": {
        "field": [
          {
            "spans": [
              {
                "end": 5,
                "start": 0,
                "type": "strong",
              },
              {
                "data": {
                  "id": "82beb55ec2f",
                  "link_type": "Media",
                },
                "end": 5,
                "start": 0,
                "type": "hyperlink",
              },
            ],
            "text": "lorem",
            "type": "paragraph",
          },
        ],
      },
      "slice_label": "Vel",
      "slice_type": "hac_habitasse",
    },
  ],
}
`;

exports[`patches link to media fields > richTextNew > static zone 1`] = `
{
  "field": [
    {
      "spans": [
        {
          "end": 5,
          "start": 0,
          "type": "strong",
        },
        {
          "data": {
            "id": "bad670dad77",
            "link_type": "Media",
          },
          "end": 5,
          "start": 0,
          "type": "hyperlink",
        },
      ],
      "text": "lorem",
      "type": "paragraph",
    },
  ],
}
`;
